timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/MultiNodePaxosTimeLockServerIntegrationTest.java

```diff
@@ -342,6 +347,43 @@ public void distinctClientsStillShareTheSameSequenceOfTimestamps() {
         assertThat(temporalSequence).isSorted();
     }
 
+    @Test
+    public void temporalOrderingIsPreservedForBatchedStartTransactionRequests() {
```

Comment:
so here, we're asserting that we got the stuff back in sorted order. But, that's not just what we do here; the contract also says that there must be an immutable lock and the results must be separated by the modulus. Fancy testing that?